### DISCUSSION TITLE
Does the station info change? If so, how often?

### DISCUSSION THREAD

**Discussion by alternapop** [weather-gov/api#746] [Q&A] [answered]
We show current weather (temperature and an icon) on some digital signs. The sign software has some built-in options but also some limitations so I want to make sure I understand the API [FAQs](https://weather-gov.github.io/api/general-faqs) before proceeding with a solution. I'd prefer not to have to create some middleware to repeatedly update the url if it's not necessary.

Using New York as an example, loading this [url](https://api.weather.gov/points/40.7756,-73.9703) gives me [this](https://api.weather.gov/gridpoints/OKX/34,38/forecast) station, "OKX/34,38".

If I understand the FAQ correctly, I can't rely on "OKX/34,38" to remain the same for those coordinates? It says the point mappings won't change "very often" but it doesn't say how often. How long should that be good for?

I also see OKX used [here](https://www.weather.gov/okx/) for NY, so I'm wondering if maybe I'm misunderstanding the FAQ and they won't change.

**Comment by sullynole** [comment]
OKX/34,38 is the forecast grid currently aligned with the lat,lon provided to points.  It may not change for years, but the WFO or grid realignment could change the grid domain one day and the grid would either be for another location or error out.  Simply verifying the geometry contains the lat,lon should provide peace of mind.

**Comment by JJKraw** [comment, marked as answer]
First, a correction. OKX/34,38 isn't a station. It's a forecast gridpoint, which is a 2.4 square km area. The second URL you list is forecast information for that area. So be clear that this isn't reported data. It's the forecast for that area.

If you go back to that first URL, look at the return value observationStations. That URL refers to actual reporting stations near the lat/lon you supply. In this case: https://api.weather.gov/gridpoints/OKX/34,38/stations

As far as the lat/lon to gridpoint mapping, my code caches it for a day. YMMV.
